tests/base/converters/test_dataframe_to_mds.py

```diff
@@ -184,3 +234,254 @@ def test_end_to_end_conversion_local(self, dataframe: Any, keep_local: bool, mer
         else:
             assert not os.path.exists(os.path.join(
                 out, 'index.json')), 'merged index is created when merge_index=False'
+
+    def test_successful_type_mapping(self):
+        spark = SparkSession.builder.getOrCreate()  # pyright: ignore
+        test_schema = StructType([
+            StructField('byte_col', ByteType(), True),
+            StructField('short_col', ShortType(), True),
+            StructField('int_col', IntegerType(), True),
+            StructField('long_col', LongType(), True),
+            StructField('float_col', FloatType(), True),
+            StructField('double_col', DoubleType(), True),
+            StructField('decimal_col', DecimalType(10, 2), True),
+            StructField('string_col', StringType(), True),
+            StructField('binary_col', BinaryType(), True),
+            StructField('array_int_col', ArrayType(IntegerType()), True),
+        ])
+        test_data = [(1, 2, 3, 4, 5.0, 6.0, Decimal('123.45'), 'eight', bytearray(b'nine'),
+                      [1, 2, 3])]
+        test_df = spark.createDataFrame(test_data, schema=test_schema)
+        expected_mappings = {
+            'byte_col': 'uint8',
+            'short_col': 'uint16',
+            'int_col': 'int32',
+            'long_col': 'int64',
+            'float_col': 'float32',
+            'double_col': 'float64',
+            'decimal_col': 'str_decimal',
+            'string_col': 'str',
+            'binary_col': 'bytes',
+            'array_int_col': 'ndarray:int32',
+        }
+        print('inferred')
```

Comment:
did you mean to leave these print statements in?

Reply:
good catch. definitely not!